avs_core/core/os/PluginManager_win32.cpp

```diff
@@ -1051,8 +1061,11 @@ bool PluginManager::TryAsAvsC(PluginFile &plugin, AVSValue *result)
   }
 
   return true;
+#else // AVS_LINUX
+  // TODO (Though I doubt anybody wants this)
```

Comment:
Actually, I think the C interface should work with relative ease.  It was already intended for building plugins with GCC (so dlsym-ing for `avisynth_c_plugin_init` should work on Linux), and the only AVS_WINDOWS define in the source of avisynth_c.cpp is for the win/linux headers.

It is true there aren't many C plugins out there, but there are two that are fairly prominent: AssRender and the C-plugin variant FFMS2 that I still keep up to date (and unless I was misunderstanding something about how the main C++ version does it, will directly use the 10/12/14-bit pix_fmts for output if given source files at those depths).  yadif could arguably be counted as well.  And on a conceptual level, the original plans for avsplus' own internal interfaces was eventually to move to a C-compatible one (it kind of fizzled out).  It wouldn't be the C interface as we currently know it, though, so it's arguable exactly what would have been needed in that situation.

Reply:
Actually, I just noticed the inline assembly for Windows x86 and assumed that it would be complex to port, although it's probably as easy as you say. I had no idea which filters were using this API either.